tools/functional-tester/etcd-runner/main.go

```diff
@@ -310,10 +320,24 @@ func runWatcher(eps []string) {
 						return
 					}
 				}
-			}(prefixes[i])
+			}(roundPrefix + "-" + prefixes[i])
 		}
 	}
 	wg.Wait()
+
+	// cancel all watch channels
+	cancelc()
+
```

Comment:
@xiang90 the channel seems to be closed on but its coming ok as true, but the value of channel being empty suggests we are receiving on a closed channel. This is the log being printed:
01:20:28.745941 expected wc to be closed, but received {{0 0 0 0} [] 0 false false <nil>}